src/scaffoldmaker/utils/tubemesh.py

```diff
@@ -281,71 +280,87 @@ def getCoordinatesFromInner(xInner, d1Inner, d2Inner, d3Inner,
+                                                            d2Surf[n + elementsCountAround],
+                                                            vector.normalise(d3Surf[n]), 0.0)
             elif n2 == elementsCountAlong:
-                curvature = interp.getCubicHermiteCurvature(xInner[n - elementsCountAround],
-                                                            d2Inner[n - elementsCountAround],
-                                                            xInner[n], d2Inner[n], vector.normalise(d3Inner[n]), 1.0)
+                curvature = interp.getCubicHermiteCurvature(xSurf[n - elementsCountAround],
+                                                            d2Surf[n - elementsCountAround],
+                                                            xSurf[n], d2Surf[n], vector.normalise(d3Surf[n]), 1.0)
             else:
                 curvature = 0.5*(
-                    interp.getCubicHermiteCurvature(xInner[n - elementsCountAround], d2Inner[n - elementsCountAround],
-                                                    xInner[n], d2Inner[n], vector.normalise(d3Inner[n]), 1.0) +
-                    interp.getCubicHermiteCurvature(xInner[n], d2Inner[n],
-                                                    xInner[n + elementsCountAround], d2Inner[n + elementsCountAround],
-                                                    vector.normalise(d3Inner[n]), 0.0))
+                    interp.getCubicHermiteCurvature(xSurf[n - elementsCountAround], d2Surf[n - elementsCountAround],
+                                                    xSurf[n], d2Surf[n], vector.normalise(d3Surf[n]), 1.0) +
+                    interp.getCubicHermiteCurvature(xSurf[n], d2Surf[n],
+                                                    xSurf[n + elementsCountAround], d2Surf[n + elementsCountAround],
+                                                    vector.normalise(d3Surf[n]), 0.0))
             curvatureAlong.append(curvature)
 
         for n3 in range(elementsCountThroughWall + 1):
             xi3 = xi3List[n3] if relativeThicknessList else 1.0/elementsCountThroughWall * n3
             for n1 in range(elementsCountAround):
                 n = n2*elementsCountAround + n1
-                norm = d3Inner[n]
-                innerx = xInner[n]
-                outerx = xOuter[n]
-                dWall = [wallThickness*c for c in norm]
+                norm = d3Surf[n]
+                surfx = xSurf[n]
+                extrudedx = xExtrudedSurf[n]
                 # x
-                x = interp.interpolateCubicHermite(innerx, dWall, outerx, dWall, xi3)
+                if outward:
+                    dWall = [wallThickness * c for c in norm]
+                    x = interp.interpolateCubicHermite(surfx, dWall, extrudedx, dWall, xi3)
+                else:
+                    dWall = [-wallThickness * c for c in norm]
+                    x = interp.interpolateCubicHermite(extrudedx, dWall, surfx, dWall, xi3)
```

Comment:
Still need separate cases for x = ~, but dWall = should not depend on inward/outward.

Reply:
Done. Kept the dWall the same, but different x.